src/inspector_agent.cc

```diff
@@ -212,17 +214,18 @@ class AsyncWriteRequest {
 
 class SetConnectedTask : public v8::Task {
  public:
-  SetConnectedTask(Agent* agent, bool connected)
-     : agent_(agent),
-       connected_(connected) {}
+  SetConnectedTask(Agent* agent): agent_(agent) {}
 
   void Run() override {
-    agent_->SetConnected(connected_);
+    int pending = agent_->SafeSetAndGetField(agent_->pending_connected_,
```

Comment:
I don't think this is right. The only way to implement it correctly is to serialize connect/disconnect messages with the actual commands, i.e. to reuse the message queue for connect/disconnect. That way input such as "<connect>, message 1, message 2, message 3, <disconnect>, <connect>, message 10, message 11, message 12, <disconnect>" would be processed in order.